pkg/prometheus/statefulset.go

```diff
@@ -857,6 +864,26 @@ func makeStatefulSetSpec(
 			}
 		}
 
+		if isWebTLS {
+			thanosArgs = append(thanosArgs, monitoringv1.Argument{Name: "prometheus.http-client-file", Value: fmt.Sprintf("%s%s", thanosSCDir, thanosSCPromClientFile)})
+			volumes = append(volumes, v1.Volume{
+				Name: "tsc-prom-client",
+				VolumeSource: v1.VolumeSource{
+					Secret: &v1.SecretVolumeSource{
+						SecretName: thanosSCPromClientSecretName(p.Name),
+					},
+				},
+			},
+			)
+			container.VolumeMounts = append(
+				container.VolumeMounts,
+				v1.VolumeMount{
+					Name:      "tsc-prom-client",
+					MountPath: thanosSCDir,
+				},
+			)
+		}
```

Comment:
I would simplify further with 2 changes:
1. use the `--prometheus.http-client` argument to avoid adding and mounting an additional secret.
2. always skip verification even if web TLS isn't configured.

 ```suggestion
		thanosArgs = append(thanosArgs, monitoringv1.Argument{Name: "prometheus.http-client", Value: `{"tls_config": {"insecure_skip_verify":true}}`})
```

Reply:
@simonpasquier ops, sorry, I forgot to mention this PR is also preparing for the [basic_auth_users](https://github.com/prometheus-operator/prometheus-operator/issues/4200) feature as well. And I also forgot we can use environment variable to inject the username and password to `--prometheus.http-client` argument. Yup, i agree with this, and will change this!